src/ui/public/timefilter/timefilter.js

```diff
@@ -23,6 +23,12 @@ uiModules
   var Events = Private(EventsProvider);
   var diff = Private(UtilsDiffTimePickerValsProvider);
 
+  config.$bind($rootScope, 'dateFormat:dow', 'dateFormat_dow');
+
+  $rootScope.$watch('dateFormat_dow', function (day) {
```

Comment:
Well, unless there's a controller that's executed on a global scale, no. This watcher needs to run on `$rootScope` (and be attached only once). I didn't know where this should go either, which is why I placed it in the service that best seemed to fit the relationship, but do advise where you'd rather move it to.